astropy/stats/circstats.py

```diff
@@ -0,0 +1,481 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This module contains simple functions for dealing with circular statistics, for
+instance, mean, variance, standard deviation, correlation coefficient, and so
+on. This module also cover tests of uniformity, e.g the Rayleigh and V0 tests.
+The Maximum Likelihood Estimator for the Von Mises distribution along with the
+Cramer-Rao Lower Bounds are also implemented. Almost all of the implementations
+are based on reference [1], which is also the basis for the R package
+'CircStats'.
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+
+
+__all__ = ['circmean','circvar', 'circmoment', 'circcorrcoef', 'rayleightest',
+           'vtest', 'vonmisesmle', 'vonmisescrlb']
+__doctest_requires__ = {'vtest': ['scipy.stats'],
+                        'vonmisescrlb': ['scipy.special']}
+
+
+def _components(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized rectangular components
+    # of the circular data.
+    data = np.asarray(data)
+
+    assert type(p) is int
+    if w is None:
+        w = np.ones(data.shape)
+    else:
+        assert w.shape == data.shape
+
+    C = np.sum(w*np.cos(p*(data-phi)), axis)/np.sum(w, axis)
+    S = np.sum(w*np.sin(p*(data-phi)), axis)/np.sum(w, axis)
+
+    return C,S
+
+def _angle(data, w=None, p=1, phi=0.0, axis=None):
+    # Utility function for computing the generalized sample mean angle
+    C,S = _components(data,w,p,phi,axis)
+    theta = np.arctan2(S,C)
+
+    mask = (S == .0)*(C == .0)
+    if mask.ndim > 0:
```

Comment:
That's correct. In case `S == .0` and `C == .0` then  `np.arctan2(S,C)` returns `0`. However, I think `np.arctan2(0,0)` equals `np.nan` makes more sense (it is also the standard in R Package CircStats). I'm replacing by `if np.any(mask)` instead.